plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java

```diff
@@ -70,10 +72,19 @@ public class GPTCompletionEngine implements DAICompletionEngine {
     private static final int MAX_REQUEST_ATTEMPTS = 3;
 
     private static final Map<String, OpenAiService> clientInstances = new HashMap<>();
-    private static final int GPT_MODEL_MAX_TOKENS = 2048;
-    private static final int MAX_PROMPT_LENGTH = 7500; // 8000 -
+    private static final int GPT_MODEL_MAX_RESPONSE_TOKENS = 2000;
     private static final boolean SUPPORTS_ATTRS = true;
 
+    private static final Pattern sizeErrorPattern = Pattern.compile(
+        "This model's maximum context length is [0-9]+ tokens, "
+            + "however you requested [0-9]+ tokens \\(([0-9]+) in your prompt; [0-9]+ "
+            + "for the completion\\)\\. Please reduce your prompt; or completion length");
+    private static final Pattern chatSizeErrorPattern = Pattern.compile(
+        "This model's maximum context length is [0-9]+ tokens\\. However, "
+            + "you requested [0-9]+ tokens \\(([0-9]+) in the messages, [0-9]+ "
+            + "in the completion\\)\\. Please reduce the length of the messages or completion\\.");
```

Comment:
They're almost the same. What's the point to keep them both?

Reply:
And I think id doesn't make sense to add trailing part (please reduce...). 
Make regex as simple as possible to match error messagem difference between two regexps is just a few words which can be replaced with `.+`